flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/dialect/JdbcDialect.java

```diff
@@ -131,8 +144,11 @@ default String getInsertIntoStatement(String tableName, String[] fieldNames) {
     }
 
     /**
-     * Get update one row statement by condition fields, default not use limit 1, because limit 1 is
-     * a sql dialect.
+     * Constructs the dialects update statement for a single row with the given condition.
+     *
+     * <p>The default implementation does not use {@code LIMIT 1} as limit is dialect specific.
+     *
+     * @return A single row update statement.
      */
     default String getUpdateStatement(
```

Comment:
The more default implementations I see here, the more I think maybe we should separate the interface and some (abstract) base implementation. Which we already have in `AbstractDialect` anyway. Should we consider moving these there?

Reply:
Or at least we shouldn't default-implement "bad" implementations (IMO), or actively document this on the interface itself ("we're giving you convenience to do little to get started, but most of it you probably want to change").